### PR TITLE
[QUESTION- FAKE PR] Questions about Property Controls usage

### PR DESCRIPTION
**Pull request by kbondanza**
Hi @hamlim,

Had a couple of questions regarding implementing the Property Controls package, added them below.

Thanks!

Screen Shot 2019-12-25 at 2 46 41 PM

### CODE REVIEW

File: packages/property-controls/README.md [hunk 1]
```diff
@@ -49,6 +49,7 @@ Ensure that this image has a high enough contrast for the color of the initials
   }
 }
 
+// For the initial state, should this be repeated for each individual component, for example if I have 3 Text Inputs and 3 Select components?
 const initialState = getInitialState(Avatar.propertyControls);
```
Comment (hamlim) [inline packages/property-controls/README.md]: Are you documenting the props of these 6 components? Or are you asking if you need to repeat this call for each of the property controls of the components?

File: packages/property-controls/README.md [hunk 2]
```diff
@@ -77,6 +78,7 @@ The Property Controls package exports the following:
 
 Rendering property control inputs is left up to the implementer to customize the rendering inputs and rendering context.
 
+Does this rendering pattern need to be seperate/repeated for each component, for example if I have a TextInput and a Select.
```
Comment (hamlim) [inline packages/property-controls/README.md]: You could share a PropertyControls component and pass it the controls for each component you are creating prop inputs for

File: packages/property-controls/README.md [hunk 3]
```diff
@@ -210,6 +212,7 @@ function PropertyControls({
       case types.shape: {
         if (isNested) {
           return (
+            // where does this PropertyControls component come from, since this function is also named Property controls (line 83)?
```
Comment (hamlim) [inline packages/property-controls/README.md]: Here we implement recursive rendering, so we render the same component we are within again. Its similar to implementing recursion for the Fibonacci sequence:

```js
function fibonacci(num) {
  if (num <= 1) return 1;

  // Call this function again, recursing here
  return fibonacci(num - 1) + fibonacci(num - 2);
} 
```

File: packages/property-controls/README.md (hunk 1 above)
Comment (kbondanza) [reply packages/property-controls/README.md]: Ah ok, I had this call in the top level of my app. So I was thinking if it needed to be repeated for each component, but seems like yes I would need to do that?

File: packages/property-controls/README.md (hunk 2 above)
Comment (kbondanza) [reply packages/property-controls/README.md]: Ok makes sense, thanks!

File: packages/property-controls/README.md (hunk 3 above)
Comment (kbondanza) [reply packages/property-controls/README.md]: Ok, makes sense, thanks Matt! Appreciate it!

File: packages/property-controls/README.md (hunk 1 above)
Comment (hamlim) [reply packages/property-controls/README.md]: Yea you would need to call this for every component being documented

File: packages/property-controls/README.md [hunk 4]
```diff
@@ -49,9 +49,12 @@ Ensure that this image has a high enough contrast for the color of the initials
   }
 }
 
+// For the initial state, should this be repeated for each individual component, for example if I have 3 Text Inputs and 3 Select components?
 const initialState = getInitialState(Avatar.propertyControls);
 
 function App() {
+  // is this backwards? should this be
+  // let [state, dispatch] = useReducer(reducer, initialState);
```
Comment (kbondanza) [inline packages/property-controls/README.md]: hi @hamlim,

I was wondering if we should update this line in the readme: `let [state, dispatch] = useReducer(initialState, reducer);` ? I was getting `Uncaught TypeError: reducer is not a function` errors when I would type in a new value for my component, so I looked up the hook useReducer in the React docs, and it looks like the reducer should be the first argument? When I switched the order of the arguments, doing `let [state, dispatch] = useReducer(reducer, initialState);`, I got it to work, just wanted to double check and I can put in a PR. 

https://reactjs.org/docs/hooks-reference.html#usereducer

Thanks!

Comment (kbondanza) [reply packages/property-controls/README.md]: Created this PR: https://github.com/hamlim/projects/pull/34

**Comment by kbondanza** [comment]
I think I'm going to close this. I have the package working in my app. I don't think I have anymore questions. Will publish my repo in a couple of days, then send you a link. Thanks! :)

**Pull request closed without merging**